### ISSUE TITLE
[Suggestion] Exclude caching based on http header value

### ISSUE THREAD

**Issue by Hashemalrifai** [filipw/AspNetWebApi-OutputCache#156]
Example: Ignore caching if Accept-Languages value is different.

**Comment by nate-fr** [comment]
You simply write a new CacheKeyGenerator by overriding MakeCacheKey in the DefaultCacheKeyGenerator class.

The function takes the context as argument and returns a string key identifying the cached ressource.

Then you simply concat the default key to the Accept-Language by accessing the context that is passed in the arguments.

Have fun.

**Comment by Hashemalrifai** [comment]
Thanks, I will try it :+1:

**Comment by ericlundmark** [comment]
Are trying to achieve the same thing have written the following cachekey
```C#
var v = base.MakeCacheKey(context, mediaType, excludeQueryString);
return context.Request.Headers.Accept + v;
```
but it does not seem to work when testing with postman. I have the following configuration
            
```C#
config.CacheOutputConfiguration().RegisterCacheOutputProvider(() => new MemoryCacheDefault());
config.CacheOutputConfiguration().RegisterDefaultCacheKeyGeneratorProvider(() => new AcceptHeaderKeyGenerator());
```

**Comment by nate-fr** [comment]
Do you really want to cache using the Accept header ? Not the AcceptLanguage ?

Because that's two different things....

**Comment by ericlundmark** [comment]
Yep, if someone requests the resource in a different format using the accept header I do not want to return the same data. E.g. someone requests an image/tiff and then requests an image/png I want to return different data, not the cached image/tiff.

**Comment by nate-fr** [comment]
Try appending this on the controller's function that you are calling : 
[CacheOutput(CacheKeyGenerator = typeof(AcceptHeaderKeyGenerator))]

**Comment by ericlundmark** [comment]
Seems to do the trick! but I expected it to be enough to configure it as the default CacheKeyGeneratorProvider? Thanks anyway :)

**Comment by Korayem** [comment]
Can you merge this PR https://github.com/filipw/Strathweb.CacheOutput/pull/126 to have this functionality out of the box

**Comment by GithubMaastricht** [comment]
Dear readers, how to enable or use "PerUserCacheKeyGenerator" ?

**1)** I have implemented

public class PerUserCacheKeyGenerator : DefaultCacheKeyGenerator
{ public override string MakeCacheKey
{ .....

**2)** and decorated the Controllers with:
[CacheOutput(ClientTimeSpan = 259200, ServerTimeSpan = 259200, CacheKeyGenerator = typeof(PerUserCacheKeyGenerator))]

**3)** and registered the PerUserCacheKeyGenerator class in WebApiConfig.cs with:

config.CacheOutputConfiguration().RegisterCacheOutputProvider(() => new MemoryCacheDefault());
config.CacheOutputConfiguration().RegisterDefaultCacheKeyGeneratorProvider(() => new PerUserCacheKeyGenerator());

But still the class PerUserCacheKeyGenerator is not used when requesting the controllers actions....

Any help?
